administration/controllers/service.go

```diff
@@ -2,72 +2,75 @@ package controllers
 
 import (
 	"encoding/json"
+	"fmt"
 	"github.com/forsam-education/cerberus/models"
 	"github.com/forsam-education/cerberus/proxy"
 	"github.com/forsam-education/cerberus/state"
 	"github.com/forsam-education/cerberus/utils"
+	"github.com/valyala/fasthttp"
 	"github.com/volatiletech/sqlboiler/boil"
 	"net/http"
 )
 
 // ListServices returns a JSON list of services.
-func ListServices(w http.ResponseWriter, _ *http.Request) {
+func ListServices(context *fasthttp.RequestCtx) {
 	services, err := models.Services().AllG()
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
+		context.Response.SetStatusCode(http.StatusInternalServerError)
 		return
 	}
-	err = json.NewEncoder(w).Encode(services)
+	err = json.NewEncoder(context.Response.BodyWriter()).Encode(services)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
+		context.Response.SetStatusCode(http.StatusInternalServerError)
 		return
 	}
 }
 
 // CreateService creates a new service from JSON request.
-func CreateService(w http.ResponseWriter, r *http.Request) {
+func CreateService(context *fasthttp.RequestCtx) {
 	var service models.Service
-	decoder := json.NewDecoder(r.Body)
-	if err := decoder.Decode(&service); err != nil {
-		w.WriteHeader(http.StatusBadRequest)
+	if err := json.Unmarshal(context.Request.Body(), &service); err != nil {
+		context.Response.SetStatusCode(http.StatusBadRequest)
 		return
 	}
+
+	message := fmt.Sprintf("There was an error Bro ! : %s", context.Request.Body())
```

Comment:
`Bro !` ?

Reply:
Also in this particular case there is no error since you handled the potential error above